scrapy/core/scheduler.py

```diff
@@ -88,6 +100,12 @@ def enqueue_request(self, request):
         return True
 
     def next_request(self):
+        delayed_request = self.dpqs.pop()
+        if delayed_request:
+            self.stats.inc_value('scheduler/dequeued/delayed/memory', spider=self.spider)
+            dqok = self._dqpush(delayed_request)
+            if not dqok:
+                self._mqpush(delayed_request)
```

Comment:
I think we may also want to reproduce the stat updates from `enqueue_request` here, as to avoid scenarios where there have been queues in disk or memory that come from delayed requests but the `enqueued` stat says 0 for disk or memory.

Maybe we could extract an `_enqueue_request` function and use it both here and in `enqueue_request`:

```python
    def _enqueue_request(self, request):
        dqok = self._dqpush(request)
        if dqok:
            self.stats.inc_value('scheduler/enqueued/disk', spider=self.spider)
        else:
            self._mqpush(request)
            self.stats.inc_value('scheduler/enqueued/memory', spider=self.spider)
```

Reply:
yeah, it makes sense